### DISCUSSION TITLE
Best practice using both cores on NXP iMX.RT1176

### DISCUSSION THREAD

**Discussion by tyalie** [zephyrproject-rtos/zephyr#50873] [Q&A] [answered]
I've read that Zephyr supports SMP and AMP but I haven't been able to find an example project for the iMX series that is able to employ both cores.

What is the current recommendation for running an applications on each core with basic synchronization between them? E.g. one core is responsible for handling a display with user input and the other one aggregates and processes data continuously

**Comment by DerekSnell** [comment, marked as answer]
Hi @tyalie ,
NXP does have an open PR https://github.com/zephyrproject-rtos/zephyr/pull/41579 to add OpenAMP support for the RT1170.  But this PR has been waiting on Zephyr's new [Sysbuild](https://docs.zephyrproject.org/latest/build/sysbuild/index.html) to leverage the multi-image build for both cores.  More details are in the PR.

In the meantime, you could leverage the support in this PR.  Best regards
